beacon-light-client/plonky2/crates/circuit_executables/bin/final_layer.rs

```diff
@@ -113,43 +122,55 @@ async fn async_main() -> Result<()> {
     >::from_bytes(commitment_proof_bytes, &commitment_data.common)?;
 
     pw.set_proof_with_pis_target(
-        &circuit_targets.commitment_mapper_circuit_targets.proof,
+        &circuit_targets.commitment_mapper_proof_target.proof,
         &commitment_final_proof,
     );
 
     pw.set_cap_target(
         &circuit_targets
-            .commitment_mapper_circuit_targets
+            .commitment_mapper_proof_target
             .verifier_circuit_target
             .constants_sigmas_cap,
         &commitment_data.verifier_only.constants_sigmas_cap,
     );
 
     pw.set_hash_target(
         circuit_targets
-            .commitment_mapper_circuit_targets
+            .commitment_mapper_proof_target
             .verifier_circuit_target
             .circuit_digest,
         commitment_data.verifier_only.circuit_digest,
     );
 
     let proof = circuit_data.prove(pw)?;
 
+    let withdrawal_credentials = balance_proof.public_inputs.withdrawal_credentials.to_vec();
+    let withdrawal_credentials = withdrawal_credentials
+        .iter()
+        .map(|credentials| hex::encode(bits_to_bytes(credentials.as_slice())))
+        .collect_vec();
+
+    // TODO: read the balance verification proof public inputs
```

Comment:
You are already doing this maybe you can remove the comment

Reply:
What I meant to say is that this is the data from redis that happen to be the same as the public inputs but we can read the public inputs directly from the proof and not rely on the redis data.